### PR TITLE
RTBA-65/nikiforov/bug-list/de sign-functionality

### PR DESCRIPTION
**Pull request by vianat**

### CODE REVIEW

File: test/specs/reg/bug-list/bug-search.js [hunk 1]
```diff
@@ -40,4 +41,90 @@ describe('bug search', function () {
...
+            let allRowsNamesInString = "";
+            for(let el of fontsize){
+                allRowsNamesInString += el.getAttribute('title');
+                allRowsNamesInString += " ";
+            }
+
+            let splitArr = allRowsNamesInString.split(" ");
+            let arrLength = splitArr.length-2;
+
+            function randomInteger(min, max) {
```
Comment (Miksyuk) [inline test/specs/reg/bug-list/bug-search.js]: may take it in helper.js?

File: test/specs/reg/bug-list/bug-search.js [hunk 2]
```diff
@@ -40,4 +41,90 @@ describe('bug search', function () {
...
+                foo.setValue("te");
+
+                let returnSearch = $$(sel.allColumnRows).length;
+
+                if (allSearchResultLength < SearchResultLength) flagA = true;
+                if (SearchResultLength < newSearchResultLength) flagB = true;
+                if (wrongSearchResult === 0) flagC = true;
+                if (returnSearch === newSearchResultLength) flagD = true;
+            }
+            if(flagA && flagB && flagC && flagD) assert.equal(true, true);
```
Comment (Miksyuk) [inline test/specs/reg/bug-list/bug-search.js]: if one from flags will be false test will fail?

Comment (Miksyuk) [reply test/specs/reg/bug-list/bug-search.js]: or will skip just?

Comment (Miksyuk) [reply test/specs/reg/bug-list/bug-search.js]: I think you need to add else assert.isTrue(true, false);

File: test/specs/reg/bug-list/bug-search.js (hunk 1 above)
Comment (vianat) [reply test/specs/reg/bug-list/bug-search.js]: thx. fix

File: test/specs/reg/bug-list/bug-search.js (hunk 2 above)
Comment (vianat) [reply test/specs/reg/bug-list/bug-search.js]: recheck and fix

File: test/selectors/filter-bar.js [hunk 3]
```diff
@@ -8,5 +8,6 @@ export default {
   tblRows: '.table tbody tr',
   tblEmpty: 'td[colspan]',
   clmnAssignee: 'td:nth-child(4)',
-  filterBar: '.input-group-append'
-}
\ No newline at end of file
+  filterBar: '.input-group-append',
```
Comment (artsenius) [inline test/selectors/filter-bar.js]: This selector is not unique
there are two elements with this class
$$('.input-group-append')

File: test/specs/reg/bug-list/bug-search.js [hunk 4]
```diff
@@ -40,4 +41,81 @@ describe('bug search', function () {
...
+            let flag = true;
+            for(let el of searchResult){
+                if( !el.getAttribute('title').includes(checkStr) ) flag = false;
+            }
+            if(flag) assert.equal(true, true);
+
+        });
+
+        it('correct search result when search field is empty', function () {
+            help.login(); // dont remove
```
Comment (artsenius) [inline test/specs/reg/bug-list/bug-search.js]: Why you need to refresh a browser?
I think you can just click "all issues"

File: test/specs/reg/bug-list/bug-search.js [hunk 5]
```diff
@@ -40,4 +41,81 @@ describe('bug search', function () {
...
+            let foo = $(sel.searchField);
+            foo.setValue(checkStr);
+
+            let searchResult = $$(sel.allColumnRows);
+
+            let flag = true;
+            for(let el of searchResult){
+                if( !el.getAttribute('title').includes(checkStr) ) flag = false;
+            }
+            if(flag) assert.equal(true, true);
```
Comment (Miksyuk) [inline test/specs/reg/bug-list/bug-search.js]: same, add "else" pls

File: test/selectors/filter-bar.js (hunk 3 above)
Comment (vianat) [reply test/selectors/filter-bar.js]: fix

File: test/specs/reg/bug-list/bug-search.js (hunk 4 above)
Comment (vianat) [reply test/specs/reg/bug-list/bug-search.js]: fix

**Pull request merged**